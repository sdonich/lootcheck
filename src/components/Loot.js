import React from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/bitcoin';

export class Loot extends React.Component {
  componentDidMount() {
    this.props.fetchBitcoin();
  }

  computeBitcoin() {
    const { bitcoin } = this.props;

    if (Object.keys(bitcoin).length === 0) return '';
    console.log(this.props.balance);
    console.log(bitcoin)

    return this.props.balance / parseInt(bitcoin.bpi.USD.rate.replace(',', ''));
  }
  render() {
    return (
      <h3>Bitcoin balance: {this.computeBitcoin()}</h3>
    );
  }
}

const mapStateToProps = state => {
  return {
    bitcoin: state.bitcoin,
    balance: state.balance
  }
}
const actionCreators = {
  fetchBitcoin
}

export default connect(mapStateToProps, actionCreators)(Loot);