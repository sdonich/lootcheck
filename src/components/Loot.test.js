import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Loot } from './Loot';

describe('Loot', () => {
  const fetchBitcoinMock = jest.fn();
  let props = { balance: 10, bitcoin: {} }
  let loot = shallow(<Loot {...props} />);
  test('render properly', () => {
    expect(toJson(loot)).toMatchSnapshot();
  });

  describe('when mounted', () => {
    beforeEach(() => {
      props.fetchBitcoin = fetchBitcoinMock;
      loot = mount(<Loot {...props} />);
    });

    test('dispatchers the `fetchBitcoin()` method it receives from props', () => {
      expect(fetchBitcoinMock).toHaveBeenCalled();
    });
  });

  describe('when there are valid bitcoin props', () => {
    beforeEach(() => {
      props = {
        balance: 10,
        bitcoin: {
          bpi: {
            USD: {
              rate: '1,000'
            }
          }
        }
      };
      loot = shallow(<Loot {...props} />);
    });
    test('displays the correct bitcoin value', () => {
      expect(loot.find('h3').text()).toBe('Bitcoin balance: 0.01');
    });
  });
});