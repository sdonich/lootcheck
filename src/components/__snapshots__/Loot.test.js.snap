// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot render properly 1`] = `
<h3>
  Bitcoin balance: 
</h3>
`;
